pandas/core/series.py

```diff
@@ -2051,18 +2051,26 @@ def map(self, arg, na_action=None):
         y : Series
             same index as caller
         """
-        values = self._values
-        if needs_i8_conversion(values.dtype):
-            boxer = i8_boxer(values)
-            values = lib.map_infer(values, boxer)
+        if is_extension_type(self.dtype):
+            values = self._values
 
-        if na_action == 'ignore':
-            mask = isnull(values)
+            if na_action == 'ignore':
+                raise NotImplementedError
 
-            def map_f(values, f):
-                return lib.map_infer_mask(values, f, mask.view(np.uint8))
+            map_f = lambda values, arg: values.map(arg)
         else:
-            map_f = lib.map_infer
+            values = _values_from_object(self)
```

Comment:
OK, let me do it separately because it is likely to need API discussion (#12741).